packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -391,11 +391,30 @@ export type SdkModelPropertyType =
   | SdkBodyParameter
   | SdkHeaderParameter;
 
+export interface MultipartOptionsType {
+  // whether name is defined in Typespec. For multipart/mixed, name may not be defined for some parts
+  isNameDefined: boolean;
+  // whether this part is for file
+  isFilePart: boolean;
+  // whether this part is multi in request payload
+  multi: boolean;
+  // relates to custom header
+  headers: SdkHeaderParameter[];
+  // undefined if filename is not set explicitly in Typespec
+  filename?: SdkModelPropertyType;
+  // undefined if contentType is not set explicitly in Typespec
+  contentType?: SdkModelPropertyType;
```

Comment:
I may ask if TCGC would calculate default contentType for us.

Reply:
Added `defaultContentTypes`. Only when `HttpPart` set, Typespec complier provides these values so TCGC doesn't calculate it but just pass it from Typespec compiler.

Reply:
I will keep it for JS for now and other languages just don't need to consume `defaultContentTypes`